modules/ximgproc/src/edge_drawing.cpp

```diff
@@ -0,0 +1,6295 @@
+
+#define pi 3.14159265
+
+    //Normalize coefficients
+    B /= A;
+    C /= A;
+    D /= A;
+    E /= A;
+    F /= A;
+    A /= A;
+
+    if (B == 0) //Then not need to rotate the axes
+    {
+        A2 = A;
+        C2 = C;
+        D2 = D;
+        E2 = E;
+        F2 = F;
+
+        //      printf("No Rotation is applied\n\n");
+    }
+    else if (B != 0) //Rotate the axes
+    {
+        rotation = true;
+
+        //Determine the rotation angle (in radians)
+        theta = atan(B / (A - C)) / 2;
+
+        //Compute the coefficients wrt the new coordinate system
+        A2 = 0.5 * (A * (1 + cos(2 * theta) + B * sin(2 * theta) + C * (1 - cos(2 * theta))));
+
+        C2 = 0.5 * (A * (1 - cos(2 * theta) - B * sin(2 * theta) + C * (1 + cos(2 * theta))));
+
+        D2 = D * cos(theta) + E * sin(theta);
+
+        E2 = -D * sin(theta) + E * cos(theta);
+
+        F2 = F;
+
+        //      printf("Rotation in degrees: %.2f\n\n", theta * 180 / pi);
+    }
+
+    //Transform the conic equation into the ellipse form
+    D3 = D2 / A2; //normalize x term's coef
+    //A3 = 1;     //A2 / A2
+
+    E3 = E2 / C2; //normalize y term's coef
+    //C3 = 1;     //C2 / C2
+
+    cX = -(D3 / 2);   //center X
+    cY = -(E3 / 2);   //center Y
+
+    F3 = A2 * pow(cX, 2.0) + C2 * pow(cY, 2.0) - F2;
+
+    //semimajor axis
+    a = sqrt(F3 / A2);
+    //semiminor axis
+    b = sqrt(F3 / C2);
+
+    //Center coordinates have to be re-transformed if rotation is applied!
+    if (rotation)
+    {
+        double tmpX = cX, tmpY = cY;
+        cX = tmpX * cos(theta) - tmpY * sin(theta);
+        cY = tmpX * sin(theta) + tmpY * cos(theta);
+    }
+
+    *pxc = cX;
+    *pyc = cY;
+
+    *pmajorAxisLength = a;
+    *pminorAxisLength = b;
+
+    //if (a > b) {
+    //    *pmajorAxisLength = a;
+    //    *pminorAxisLength = b;
+    //}q
+    //else {
+    //    *pmajorAxisLength = b;
+    //    *pminorAxisLength = a;
+    //}
+
+    return theta;
+#undef pi
+}
+
+// ---------------------------------------------------------------------------
+// Given an ellipse equation, computes "noPoints" many consecutive points
+// on the ellipse periferi. These points can be used to draw the ellipse
+// noPoints must be an even number.
+//
+void EdgeDrawingImpl::ComputeEllipsePoints(double* pvec, double* px, double* py, int noPoints)
+{
+    if (noPoints % 2)
+        noPoints--;
+    int npts = noPoints / 2;
+
+    double** u = AllocateMatrix(3, npts + 1);
+    double** Aiu = AllocateMatrix(3, npts + 1);
+    double** L = AllocateMatrix(3, npts + 1);
+    double** B = AllocateMatrix(3, npts + 1);
+    double** Xpos = AllocateMatrix(3, npts + 1);
+    double** Xneg = AllocateMatrix(3, npts + 1);
+    double** ss1 = AllocateMatrix(3, npts + 1);
+    double** ss2 = AllocateMatrix(3, npts + 1);
+    double* lambda = new double[npts + 1];
+    double** uAiu = AllocateMatrix(3, npts + 1);
+    double** A = AllocateMatrix(3, 3);
+    double** Ai = AllocateMatrix(3, 3);
```

Comment:
`Matx33d`?

Reply:
hi @alalek i believe the code is almost ready to merge for testing the algorithms. There are many work to do such as writing documentation and polishing API design and adding some more algorithms from ED_Lib. I will work to evaluate but i believe it will be fine if it merged.